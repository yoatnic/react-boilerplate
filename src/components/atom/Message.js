import React from "react";

const Message = props => {
  const style = {
    width: "100%",
    borderBottom: "1px solid"
  };
  return <div style={style}>{props.message}</div>;
};

export default Message;
