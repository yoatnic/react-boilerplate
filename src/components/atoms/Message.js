//@flow
import React from "react";
import PropTypes from "prop-types";

type Props = {
  message: string
};

const Message = (props: Props) => {
  const style = {
    width: "100%",
    borderBottom: "1px solid"
  };
  return <div style={style}>{props.message}</div>;
};

export default Message;
