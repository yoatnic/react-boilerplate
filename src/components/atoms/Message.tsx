import * as React from "react";
import styled from "styled-components";

interface Props {
  message: string;
}

const MessageContent = styled.div`
  width: "100%";
  border-bottom: "1px solid";
`;

const Message = (props: Props) => {
  return <MessageContent>{props.message}</MessageContent>;
};

export default Message;
