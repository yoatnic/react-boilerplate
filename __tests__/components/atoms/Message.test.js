import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import Message from "components/atoms/Message";

test("show message", () => {
  const tree = renderer.create(<Message message="test message" />).toJSON();
  expect(tree.children[0]).toEqual("test message");
});

test("snap shot", () => {
  const tree = renderer.create(<Message message="test message" />).toJSON();
  expect(tree).toMatchSnapshot();
});
