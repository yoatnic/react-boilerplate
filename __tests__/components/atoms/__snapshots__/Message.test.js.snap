// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snap shot 1`] = `
.c0 {
  width: "100%";
  border-bottom: "1px solid";
}

<div
  className="c0"
>
  test message
</div>
`;
